test/metabase/query_processor/middleware/resolve_joins_test.clj

```diff
@@ -99,17 +99,17 @@
                               :condition    [:= $category_id &c.categories.id]
                               :fields       [&c.categories.name]}]}))))))
 
-(deftest join-same-table-twice-test
-  (testing "Does joining the same table twice without an explicit alias give both joins unique aliases?"
+(deftest join-table-without-alias-test
+  (testing "Does joining a table an explicit alias add a default alias?"
     (is (= (mt/mbql-query venues
              {:joins        [{:source-table $$categories
-                              :alias        "source"
+                              :alias        "__join"
                               :strategy     :left-join
                               :condition    [:= $category_id 1]}
                              {:source-table $$categories
-                              :alias        "source_2"
+                              :alias        "__join"
```

Comment:
i'm confused why the diff before had two aliases "source" and "source_2" but now we end up with only one alias "__join"

Reply:
and running this in the repl i get:

```clojure
resolve-joins-test=> (qp/query->native
                       (resolve-joins
                         (mt/mbql-query venues
                                        {:joins [{:source-table $$categories
                                                  :condition    [:= $category_id 1]}
                                                 {:source-table $$categories
                                                  :condition    [:= $category_id 2]}]})))
Execution error (ExceptionInfo) at schema.core/validator$fn (core.clj:155).
Value does not match schema: {:query {:joins (not ("All join aliases must be unique." a-clojure.lang.PersistentVector))}}
```

Which is what seems like is happening here. If this constraint was removed, we would end up with something like

```sql
select venues.*, __joins.*, __joins.* from ...
```
which seems like it would be logically incorrect and also fail the validation in the driver. Am i missing something?